pmd-cataloger/src/main/java/sfdc/sfdx/scanner/xml/XmlReader.java

```diff
@@ -26,10 +30,22 @@ public static XmlReader getInstance() {
 	 *
 	 * @param path - The path to a resource.
 	 * @return - An InputStream for the provided resource.
+	 * @throws FileNotFoundException - if path is a regular file but that file can't be read
 	 */
-	private InputStream getResourceAsStream(String path) {
-		final InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
-		return in == null ? getClass().getResourceAsStream(path) : in;
+	private InputStream getResourceAsStream(String path) throws FileNotFoundException {
+		InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
```

Comment:
As someone who's not terribly familiar with IO in Java, I'd love to see some comments explaining what's going on here.

Reply:
Good question. Let me see if I can find out why the code is looking at two class loaders.